qutebrowser/misc/debugcachestats.py

```diff
@@ -0,0 +1,50 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2019-2019 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Implementation of the command debug-cache-stats.
+
+Because many modules depend on this command, this needs to have as few
+dependencies as possible to avoid cyclic dependencies.
+"""
+
+import typing
+from typing import TypeVar, Callable
+
+
+# The second element of each tuple should be a lru_cache wrapped function
+_CACHE_FUNCTIONS = []  # type: typing.List[typing.Tuple[str, typing.Any]]
+
+
+_T = TypeVar('_T', bound=Callable)
+
+
+def register(name: typing.Optional[str] = None) -> Callable[[_T], _T]:
+    """Register a lru_cache wrapped function to be used with debug_cache_stats.
+    """
```

Comment:
Flake8 complain at this line, although I have no idea how to fix it.

Reply:
"""
    Register a lru_cache wrapped function to be used with debug_cache_stats.
    """
Probably. Putting it all on one line might work but I'm not sure the new line length limits apply to docstrings.

Reply:
I'd just `s/to be used with/for/` and then the line is short enough to fit it all in one line :wink:

Reply:
But no other docstrings are formatted like that.